podcast-player/src/App.js

```diff
@@ -27,10 +28,13 @@ const PODCASTS = [
 ];
 
 function App() {
+  const [sideBarOpen, setSideBarOpen] = useState(true);
   const [shows, setShows] = useState([]);
```

Comment:
Ah, just now seeing this as well. `show` is derived state, and it would be simpler and faster to just compute the show on-demand.

```js
const show = shows.find(show => show.id === showId);
```

Reply:
Im not quite sure how to how to compute this on demand, as its passed into components and espisodeLIst is conditionally shown if its set or not